app/main/posts/views/post-filters.service.js

```diff
@@ -250,7 +262,7 @@ function PostFiltersService(_, FormEndpoint, TagEndpoint, $q) {
         return !_.isEmpty(getActiveFilters(filters));
     }
 
-    function setMode(newMode, id) {
+    function setMode(newMode, obj) {
```

Comment:
setting the obj instead of the ID makes sense to avoid unnecessary calls to the backend (nothing wrong with calling it, but if we have the object, it's alright to set it here IMO and gain the access to filters directly.)